Wikipedia/Code/WMFImageURLParsing.m

```diff
@@ -6,7 +7,7 @@
     dispatch_once(&onceToken, ^{
         // TODO: try to read serialized regex from disk to prevent needless pattern compilation on next app run
         NSError* patternCompilationError;
-        imageNameFromURLRegex = [NSRegularExpression regularExpressionWithPattern:@"^\\d+px-([^.]*(\\.[^.$]{4,})*\\.[^.]*).*$"
+        imageNameFromURLRegex = [NSRegularExpression regularExpressionWithPattern:@"^\\d+px-(.*)"
```

Comment:
You can also easily [write a test case](https://github.com/wikimedia/wikipedia-ios/pull/419/files#diff-a128dce327b7690819fa2406bb2da2c5R46) in the Android app to see if your parsing code has this issue.